crates/install-wheel-rs/src/linker.rs

```diff
@@ -249,6 +252,29 @@ fn clone_wheel_files(
     let mut count = 0usize;
     let mut attempt = Attempt::default();
 
+    if cfg!(target_os = "macos") {
+        // On macOS, the directory mtime is not updated when cloning occurs and the mtime is used
+        // by CPython's import mechanisms to determine if it should look for new packages in a directory.
+        // Here, we force the mtime to be updated to ensure that packages are importable without
+        // manual cache invalidation.
+        //
+        // <https://github.com/python/cpython/blob/8336cb2b6f428246803b02a4e97fce49d0bb1e09/Lib/importlib/_bootstrap_external.py#L1601>
+
+        // Construct a unique file name
+        let mut hasher = FxHasher::default();
+        hasher.write(wheel.as_ref().as_os_str().as_bytes());
+        let mtime_file = site_packages
+            .as_ref()
+            .join(format!("uv-mtime-{}", hasher.finish()));
+
+        // Create and remove the file
+        fs_err::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(mtime_file.clone())?;
+        fs_err::remove_file(mtime_file)?;
```

Comment:
Arguably we could avoid failing on these, but it seems nice to know if we're leaving them around due to some failure.

Reply:
I'm looking into that next yeah, https://doc.rust-lang.org/std/fs/struct.File.html#method.set_modified also

Reply:
I thought that _may_ not work for directories but I didn't test it. I only tested `filetime` crate.

Reply:
https://github.com/astral-sh/uv/pull/2545/commits/205b224ac5a475618520c1ea8aeeff9703eb677b